### PR TITLE
#2317223 | Port 'Block user' action to D8

### PR DESCRIPTION
**Pull request by M1r1k**
https://www.drupal.org/node/2317223

### CODE REVIEW

File: tests/src/Action/BlockUserTest.php [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    $this->userStorage = $this->getMock('Drupal\user\UserStorageInterface');
+    $this->sessionManager = $this->getMock('Drupal\Core\Session\SessionManagerInterface');
+
+    $this->action = new BlockUser([], '', ['context' => [
```
Comment (fago) [inline tests/src/Action/BlockUserTest.php]: Please update to RulesIntegrationTestBase and use the manager to create it.

File: tests/src/Action/BlockUserTest.php [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  public function testBlockUserWithBlockedAuthenticatedUser() {
+    $user = $this->getUserMock(FALSE);
+
+    $user->expects($this->never())
+      ->method('block');
+
+    $this->sessionManager->expects($this->never())
+      ->method('delete');
+
+    $this->action->setContextValue('user', $this->getMockTypedData($user));
```
Comment (fago) [inline tests/src/Action/BlockUserTest.php]: This should pass just $user, i.e. remove the call to getMockedTypedData(). See https://www.drupal.org/node/2317353#comment-9037793

File: tests/src/Action/BlockUserTest.php (hunk 1 above)
Comment (fago) [inline tests/src/Action/BlockUserTest.php]: That needs to be addressed still.

File: src/Plugin/Action/BlockUser.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+   * Constructs a BlockUser object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param SessionManagerInterface $session_manager
+   *   The alias storage service.
```
Comment (klausi) [inline src/Plugin/Action/BlockUser.php:51]: Alias storage? Wrong comment.

File: src/Plugin/Action/BlockUser.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+  /**
+   * Executes the plugin.
+   */
+  public function execute() {
+    /* @var $user \Drupal\user\Entity\User */
+    $user = $this->getContextValue('user');
+
+    // Do nothing if user is already disable or anonymous.
```
Comment (klausi) [inline src/Plugin/Action/BlockUser.php:84]: Should be "disabled".

File: src/Plugin/Action/BlockUser.php [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    if ($user->isActive() && $user->isAuthenticated()) {
+      $user->block();
+      $this->sessionManager->delete($user->id());
+    }
+  }
+
+}
```
Comment (klausi) [inline src/Plugin/Action/BlockUser.php:91]: Missing line break.

File: tests/src/Action/BlockUserTest.php [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+      ->will($this->returnValue($active));
+
+    $user->expects($this->exactly((int) $active))
+      ->method('isAuthenticated')
+      ->will($this->returnValue($authenticated));
+
+    return $user;
+  }
+
+}
```
Comment (klausi) [inline tests/src/Action/BlockUserTest.php:161]: Missing line break.

File: tests/src/RulesIntegrationTestBase.php [hunk 7]
```diff
@@ -30,6 +30,11 @@
 abstract class RulesIntegrationTestBase extends RulesUnitTestBase {
 
   /**
+   * @var \Symfony\Component\DependencyInjection\ContainerInterface
```
Comment (klausi) [inline tests/src/RulesIntegrationTestBase.php:33]: Comment is missing, something like "The services container used during testing."

File: tests/src/Action/BlockUserTest.php [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+      ->method('delete')
+      ->with('123');
+
+    $this->action->setContextValue('user', $user);
+
+    $this->action->execute();
+  }
+
+  /**
+   * Creates a mock node.
```
Comment (klausi) [inline tests/src/Action/BlockUserTest.php:137]: Wrong comment, this is a mock user.

**Comment by klausi** [comment]
Continued in #156

**Comment by klausi** [comment]
Oops, nope, that's a different one.

**Comment by klausi** [comment]
Continued in #161

**Pull request closed without merging**